File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:

        # Create an empty dictionary to store character counts
        char_count = {}

        # Iterate through each word in the list
        for element in words:

            # Iterate through each character in the word
            for char in element:
                
                # Update the character count in the dictionary
                if char in char_count:
                    char_count[char] += 1
                else:
                    char_count[char] = 1

        # Extract the list of character counts from the dictionary
        char_count_list = char_count.values()

        # Check if it is possible to make all strings equal
        for num in char_count_list:
            # If the count is not divisible by the total number of words, return False
            if num % len(words) != 0:
                return False

        # If all character counts are divisible by the total number of words, return True
        return True
        
        

File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.js
/**
 * @param {string[]} words
 * @return {boolean}
 */
var makeEqual = function(words) {

    // Initialize an empty object to store character counts.
    let count = {};

    // Iterate through each word in the array.
    for (let element of words) {

        // Iterate through each character in the current word.
        for (let char of element) {
            
            // Update the count of the current character in the object.
            if (char in count) {
                count[char] += 1;
            } else {
                count[char] = 1;
            }
        }
    }

    // Get a list of all character occurrences.
    let count_list = Object.values(count);

    // Iterate through the counts and check if they are divisible by the number of words.
    for (let num of count_list) {
        // If the count is not divisible by the number of words, return false.
        if (num % words.length !== 0) {
            return false;
        }
    }

    // If all counts are divisible by the number of words, return true.
    return true;
};